File: ByLanguage/android/Cookbook/AndroidLifecycle/app/src/main/java/com/tq/activitylifecycle/MainActivity.java
package com.tq.activitylifecycle;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

        @Override
        protected void onCreate(Bundle savedInstanceState) {
                super.onCreate(savedInstanceState);
                setContentView(R.layout.activity_main);
                toast("onCreate");
        }

        private void toast(String message) {
                Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
        }

        @Override
        protected void onStart() {
                toast("onStart");
                super.onStart();
        }

        @Override
        protected void onResume() {
                toast("onResume");
                super.onResume();
        }

        @Override
        protected void onRestart() {
                toast("onRestart");
                super.onRestart();
        }

        @Override
        protected void onPause() {
                toast("onPause");
                super.onPause();
        }

        @Override
        protected void onStop() {
                toast("onStop");
                super.onStop();
        }

        @Override
        protected void onDestroy() {
                toast("onDestroy");
                super.onDestroy();
        }

}

File: ByLanguage/android/DataBindingDemo/app/src/main/java/com/tq/databingdingdemo/MainActivity.kt
package com.tq.databingdingdemo

import android.os.Bundle
import android.view.View

import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil

import com.tq.databingdingdemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.message.text = "DataBindingDemo"
    }
}
